lib/iris/analysis/trajectory.py

```diff
@@ -133,6 +133,55 @@ def __repr__(self):
         return 'Trajectory(%s, sample_count=%s)' % (self.waypoints,
                                                     self.sample_count)
 
+    def _get_interp_points(self):
+        """
+        Translate `self.sampled_points` to the format expected by the
+        interpolator. If the CRS of the sample points does not match the CRS
+        of the cube to interpolate, also transform the points.
+
+        Returns:
+            `self.sampled points` in the format required by
+            `:func:`~iris.analysis.trajectory.interpolate`.
+
+        """
+        points = {k: [point_dict[k] for point_dict in self.sampled_points]
+                  for k in self.sampled_points[0].keys()}
+        return [(k, v) for k, v in points.items()]
+
+    def interpolate(self, cube, method=None):
+        """
+        Calls :func:`~iris.analysis.trajectory.interpolate` to interpolate
+        ``cube`` on the defined trajectory.
+
+        Assumes that the coordinate names supplied in the waypoints
+        dictionaries match to coordinate names in `cube`, and that points are
+        supplied in the same coord_system as in `cube`, where appropriate (i.e.
+        for horizontal coordinate points).
+
+        Args:
+
+        * cube
+             The source Cube to interpolate.
+
+        Kwargs:
+
+        * method:
+            The interpolation method to use; "linear" (default) or "nearest".
+            Only nearest is available when specifying multi-dimensional
+            coordinates.
+
+        """
+        sample_points = self._get_interp_points()
+        interpolated_cube = interpolate(cube, sample_points, method=method)
+        # Add an "index" coord to name the anonymous dimension produced by
+        # the interpolation, if present.
+        if len(interpolated_cube.dim_coords) < interpolated_cube.ndim:
+            index_coord = iris.coords.DimCoord(range(self.sample_count),
+                                               long_name='index')
+            anon_dim_index = interpolated_cube.shape.index(self.sample_count)
+            interpolated_cube.add_dim_coord(index_coord, anon_dim_index)
```

Comment:
Bit worried that this might be a problem for cubes which already have dimensionless coordinates as dim coords.  

If the original dimensionless coordinates are processed and thus consumed in the production of trajectory points, this will not be a problem.  Also, bearing in mind that this method checks the _interpolated_ cube (not the original cube) for new anonymous dimensions, this would only be a problem if the waypoints were passed in as anonymous coords, which seems a bit stupid anyway.

Considering all the above, I am happy with this process.